src/credentials/auth.ts

```diff
@@ -902,15 +902,64 @@ export async function createStartUrlPrompter(title: string, ignoreScopes = true)
 }
 
 export async function createBuilderIdConnection(auth: Auth) {
+    const existingConn = await promptLogoutExistingBuilderIdConnection(auth)
+
     const newProfile = createBuilderIdProfile()
-    const existingConn = (await auth.listConnections()).find(isBuilderIdConnection)
     if (existingConn && !hasScopes(existingConn, newProfile.scopes)) {
         return migrateBuilderId(auth, existingConn, newProfile)
     }
 
     return existingConn ?? (await auth.createConnection(newProfile))
 }
 
+/**
+ * Prompts the user to log out of an existing Builder ID connection if it exists.
+ *
+ * @returns The existing Builder ID connection if it exists and the user does not want to log out.
+ *
+ *          Otherwise, undefined, due to user selecting logout or the connection never existing
+ *          in the first place.
+ */
+async function promptLogoutExistingBuilderIdConnection(auth: Auth): Promise<SsoConnection | undefined> {
+    const builderIdConn = (await auth.listConnections()).find(isBuilderIdConnection)
+
+    if (!builderIdConn) {
+        return undefined
+    }
+
+    const items: DataQuickPickItem<string>[] = [
+        {
+            data: 'signout',
+            label: `You already signed in with an ${getIdeProperties().company} Builder ID. Sign out to add another?`,
+            detail: `This will sign out of your current ${
+                getIdeProperties().company
+            } Builder ID and open the sign-in page in browser.`,
+        },
+        { data: 'cancel', label: 'Cancel' },
+    ]
+    const resp = await showQuickPick(items, {
+        title: `Sign in to different ${getIdeProperties().company} Builder ID`,
+        buttons: createCommonButtons() as vscode.QuickInputButton[],
+    })
+
+    switch (resp) {
+        case 'signout': {
+            await signout(auth, builderIdConn)
+
+            // Sanity check signout worked
+            const sanityCheckConn = (await auth.listConnections()).find(isBuilderIdConnection)
+            if (sanityCheckConn !== undefined) {
+                throw new Error('Failed to sign out of Builder ID.')
+            }
+
+            return undefined
+        }
+        default: {
+            return builderIdConn
+        }
```

Comment:
Throwing an error is more explicit. Returning the connection causes an error to get thrown later, might as well fail early with more appropriate semantics:
```suggestion
        default: {
            throw new CancellationError('user')
        }
```

Reply:
Maybe I'm misunderstanding but only the function `createBuilderIdConnection()` calls this function. Returning the `builderIdConn` is intentional since `createBuilderIdConnection()` sometimes expects a connection from the call.